SmartDeviceLink/SDLFile.m

```diff
@@ -100,17 +100,17 @@ - (NSData *)data {
 /**
  Initalizes a socket from which to read data.
 
+ @discussion A new `NSInputStream` is created when requested instead of returning the already open `NSInputStream`. This is done because once a opened, a stream *cannot* be closed and reopened. If the same file is accessed multiple times (i.e. a non-persistant file is uploaded before and after a disconnect from Core during the same session) the file cannot be accessed after the first access because the stream is closed once the upload is completed. Apple `NSInputStream` doc: https://developer.apple.com/documentation/foundation/nsstream/1411963-open
+
  @return A socket
  */
 - (NSInputStream *)inputStream {
-    if (!_inputStream) {
-        if (_fileURL) {
-            // Data in file
-            _inputStream = [[NSInputStream alloc] initWithURL:_fileURL];
-        } else if (_data.length != 0) {
-            // Data in memory
-            _inputStream = [[NSInputStream alloc] initWithData:_data];
-        }
+    if (_fileURL) {
```

Comment:
I don't think it makes sense to keep the stream open for all  files as the only time a file is going to be uploaded again is if the upload to Core fails or if the user reconnects to Core during the same session. Otherwise there are going to be a bunch of unnecessary open input streams.

Reply:
I would recommend removing the corresponding property since the ivar is never referenced outside this method and the property isn't needed.

I also think it would be clearer to rename the method so the intention is clearer:
```
- (NSInputStream *)openInputStream {
    NSAssert(_fileURL != nil && _data.length != 0, @"Unable to open input stream");
    NSInputStream* inputStream = nil;
    if (_fileURL) {
        // Data in file
        inputStream = [[NSInputStream alloc] initWithURL:_fileURL];
    } else if (_data.length != 0) {
        // Data in memory
        inputStream = [[NSInputStream alloc] initWithData:_data];
    }
    [inputStream open];
    return inputStream;
}
```

SDLUploadFileOperation can then be simplified like this:

```
- (NSInputStream *)sdl_openInputStreamWithFile:(SDLFile *)file {
    return [file openInputStream];
} 
```
Or ideally that method can be completely removed (if symmetry isn't deemed important) since the SDLFile now provides the openInputStream method itself.